src/App.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import Board from './components/Board';
+
+// Any alive cell that is touching less than two alive neighbours dies.
+// Any alive cell touching four or more alive neighbours dies.
+// Any alive cell touching two or three alive neighbours does nothing.
+// Any dead cell touching exactly three alive neighbours becomes alive.
+
+//number of rows/columns
+const size = 40;
+
+//cellInitialStatus returns random true/false value to identify alive/dead statuses
+//and it's the default value for the argument
+const generateBoardStatus = (
+	cellInitialStatus = () => Math.random() >= 0.6
+) => {
+	const grid = [];
+	//generating the rows
+	for (let row = 0; row < size; row++) {
+		grid[row] = [];
+		//generating the columns and associating the cell status with it
+		for (let column = 0; column < size; column++) {
+			grid[row][column] = cellInitialStatus();
+		}
+	} //end of for loop
+
+	return grid;
+};
+
+function App() {
+	//creating state for boardStatus to be passed as props
+	const [boardStatus, setBoardStatus] = useState(generateBoardStatus());
+
+	//boolean for start/pause
+	const [runGame, setRunGame] = useState(false);
+
+	//takes the index for row/column when called and evaluates if
+	//the cell is on the board and reduce the neighbor cells to an array only with cells that are true
+	const aliveCells = (row, column) => {
+		//checking only the 4 imminent "neighbour" cells
+		const neighbourCells = [
+			[-1, -1],
+			[-1, 0],
+			[-1, 1],
+			[0, 1],
+			[1, 1],
+			[1, 0],
+			[1, -1],
+			[0, -1],
+		];
+
+		return neighbourCells.reduce((aliveCells, cells) => {
+			const rowPosition = row + cells[0];
+			const columnPosition = column + cells[1];
+
+			const isCellOnBoard =
+				rowPosition >= 0 &&
+				rowPosition < size &&
+				columnPosition >= 0 &&
+				columnPosition < size;
+
+			if (
+				aliveCells < 4 &&
+				isCellOnBoard &&
+				boardStatus[rowPosition][columnPosition]
+			) {
+				return aliveCells + 1;
+			} else {
+				return aliveCells;
+			}
+		}, 0);
+	}; //closing function alive cells
+
+	const updateBoard = () => {
+		//having to do this due to JS not allowing clone of multidimensional arrays
+		let clonedBoard = JSON.parse(JSON.stringify(boardStatus));
+
+		for (let r = 0; r < size; r++) {
+			for (let c = 0; c < size; c++) {
+				const totalAliveCells = aliveCells(r, c);
```

Comment:
use row and column instead of just r and c is more readable, remember that you are not just writing code to the computer but to people